src/run/run.c

```diff
@@ -1356,6 +1381,24 @@ static int make_unit_name(sd_bus *bus, UnitType t, char **ret) {
         assert(t < _UNIT_TYPE_MAX);
         assert(ret);
 
+        /* Preferable use our PID + pidfd ID as identifier, if available. It's a boot time unique identifier
+         * managed by the kernel. Unfortunately only new kernels support this, hence we keep a lot of
+         * fallback logic in place. */
```

Comment:
So I think we should utterly drop the whole dbus identifier stuff. It creates extra churn when soft-reboot is used and such, so why not always work with uuid when pidfd id is unavailable?

Reply:
good point.